src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -133,6 +148,9 @@ public void init() {
         	// High level, so check all results are in the expected context
         	return hca.getHtmlContexts(attack, targetContext, ignoreFlags);
         }
+        if (findDecoded) {
+            return hca.getHtmlContexts(getURLDecode(attack));
+        }
         return hca.getHtmlContexts(attack);

```

Comment:
`hca.getHtmlContexts(findDecoded ? getURLDecode(attack) : attack)` and remove the above if?